### ISSUE TITLE
Generate different forecast result at different run

### ISSUE THREAD

**Issue by yug95** [tensorflow/probability#770]
While running tfp sts model, i noticed each time i ran model it generated different forecast which is kind off problematic for inference.

```
import tensorflow_probability as tfp
trend = tfp.sts.LocalLinearTrend(observed_time_series=co2_by_month)
seasonal = tfp.sts.Seasonal(
    num_seasons=12, observed_time_series=co2_by_month)
model = tfp.sts.Sum([trend, seasonal], observed_time_series=co2_by_month)


co2_model = build_model(co2_by_month_training_data)
variational_posteriors = tfp.sts.build_factored_surrogate_posterior(
    model=co2_model)

num_variational_steps = 200 
num_variational_steps = int(num_variational_steps)

optimizer = tf.optimizers.Adam(learning_rate=.1)

q_samples_co2_ = variational_posteriors.sample(50)


co2_forecast_dist = tfp.sts.forecast(
    co2_model,
    observed_time_series=co2_by_month_training_data,
    parameter_samples=q_samples_co2_,
    num_steps_forecast=num_forecast_steps)


```

Here i have taken `mean` value. How to give `Seed` like we give in Random Forest algorithm or is there any other way to fix ?

**Comment by davmre** [comment]
There are two sources of randomness here: first in the initialization of
the variational optimization, and then in sampling parameters from the
optimized variational posterior. You can pass a seed to both of those
operations:

variational_posteriors = tfp.sts.build_factored_surrogate_posterior(
    model=co2_model, seed=seed)

...

q_samples_co2_ = variational_posteriors.sample(50, seed=seed)

To get deterministic results in eager mode (where each op is
implicitly in its own 'graph' context with a random graph-level seed)
you likely also need to call tf.random.set_seed(seed) at the top of
your code.

**Comment by yug95** [comment]
Thanks @davmre :)

**Comment by yug95** [comment]
@davmre bychance you got a time to look into this ? please let me know

**Comment by davmre** [comment]
Did you try setting the graph-level seed (`tf.random.set_seed(seed_value)`) right before you create each STS model?

If you can reproduce the issue in a minimal example with a single time series, and get it running in a [colab](colab.research.google.com), I can try to dig further.

**Comment by yug95** [comment]
Thanks @davmre its working now.
